src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1221,6 +1221,44 @@ void MPS::full_state_vector_internal(cvector_t& statevector,
   statevector = reverse_all_bits(temp_statevector, num_qubits);
 }
 
+void MPS::get_amplitude_vector(const reg_t &base_values, cvector_t &amplitude_vector) {
+  move_all_qubits_to_sorted_ordering();
+  uint_t num_values = base_values.size();
+  std::string base_value;
+#pragma omp parallel for if (num_values > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
+  for (int_t i=0; i<static_cast<int_t>(num_values); i++) {
```

Comment:
Nicer:
```
for (const auto &bv_int : base_values)
```

Reply:
I agree, but under omp parallel I must use an "int" as the iterator of the loop. Compilation fails when I try your suggestion.